sources/packages/api/src/route-controllers/institution/institution.aest.controller.ts

```diff
@@ -239,15 +246,85 @@ export class InstitutionAESTController extends BaseController {
    * @param payload authorizations to be associated with the user.
    * @returns Primary identifier of the created resource.
    */
-  // TODO: Add API Responses and validations. This method will be worked in the upcoming PR.
+  @ApiNotFoundResponse({ description: "Institution not found." })
+  @ApiUnprocessableEntityResponse({
+    description:
+      "User to be added was not found on BCeID Account Service " +
+      "or the user does not belong to the same institution " +
+      "or the user already exists " +
+      "or a second legal signing authority is trying to be set when one is already in place.",
+  })
   @Post(":institutionId/user")
   async createInstitutionUserWithAuth(
-    institutionId: number,
+    @Param("institutionId", ParseIntPipe) institutionId: number,
     @Body() payload: CreateInstitutionUserAPIInDTO,
   ): Promise<PrimaryIdentifierAPIOutDTO> {
     return this.institutionControllerService.createInstitutionUserWithAuth(
       institutionId,
       payload,
     );
   }
+
+  /**
+   * Update the user authorizations for the institution user.
+   * @param userName user to have the permissions updated.
+   * @param payload permissions to be updated.
+   */
+  @ApiNotFoundResponse({
+    description: "User to be updated not found.",
+  })
+  @ApiUnprocessableEntityResponse({
+    description:
+      "The user is not active" +
+      " or the user permission is being updated in a way that no admin will be present" +
+      " or a second legal signing authority is trying to be set and only one is allowed.",
+  })
+  @Patch("user/:userName")
+  async updateInstitutionUserWithAuth(
+    @Param("userName") userName: string,
```

Comment:
I see that the service is using the userName to retrieve the institution  user from institution_users table. 
Not a part of this PR, we can use the institutionUserId instead for this API right?

Reply:
Yes, I noticed that and I do not remember the reason why it was built like this. I would be happy to take care of it in the upcoming PR. The `userName` is a unique column so no big worries to keep or change, I would like to change to be more aligned with what we have been doing.
@ann-aot @henryoforeh-aot @guru-aot any concern about it?

Reply:
mm, I am also not remembering the reason why we used `userName`.

Reply:
For this scenario, I think we can use the institution user rather using the username.